src/vm/gccover.cpp

```diff
@@ -1371,72 +1479,69 @@ void DoGcStress (PCONTEXT regs, MethodDesc *pMD)
 
     Thread *pThread = GetThread();
 
+    if (!IsGcCoveregeInterruptInstruction((LPVOID)instrPtr))
+    {
+        _ASSERTE(IsOriginalInstruction((LPVOID)instrPtr, gcCover, offset));
+        // Someone beat us to it, just go on running.
+        return;
+    }
+
+    bool atCall;
+    bool afterCallProtect[2] = { 0 };
+
 #if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
 
     BYTE instrVal = *(BYTE *)instrPtr;
     forceStack[6] = &instrVal;            // This is so I can see it fastchecked
-    
-    if (instrVal != INTERRUPT_INSTR && 
-        instrVal != INTERRUPT_INSTR_CALL && 
-        instrVal != INTERRUPT_INSTR_PROTECT_RET) {
-        _ASSERTE(instrVal == gcCover->savedCode[offset]);  // someone beat us to it.
-        return;       // Someone beat us to it, just go on running
-    }
-
-    bool atCall = (instrVal == INTERRUPT_INSTR_CALL);
-    bool afterCallProtect = (instrVal == INTERRUPT_INSTR_PROTECT_RET);
 
-#elif defined(_TARGET_ARM_)
+    atCall = (instrVal == INTERRUPT_INSTR_CALL);
 
-    WORD instrVal = *(WORD*)instrPtr;
-    forceStack[6] = &instrVal;            // This is so I can see it fastchecked
+    if (instrVal == INTERRUPT_INSTR_PROTECT_BOTH_RET)
+    {
+        afterCallProtect[0] = afterCallProtect[1] = true;
+    }
+    else if (instrVal == INTERRUPT_INSTR_PROTECT_FIRST_RET)
+    {
+        afterCallProtect[0] = true;
+        afterCallProtect[1] = false;
+    }
+    else if (instrVal == INTERRUPT_INSTR_PROTECT_SECOND_RET)
+    {
+        afterCallProtect[0] = false;
+        afterCallProtect[1] = true;
+    }
+    else
+    {
+        afterCallProtect[0] = afterCallProtect[1] = false;
+    }
 
-    size_t instrLen = GetARMInstructionLength(instrVal);
+#elif defined(_TARGET_ARM_)
+    forceStack[6] = (WORD*)instrPtr;            // This is so I can see it fastchecked
```

Comment:
I think you want to leave this as `&instrVal`.

The comment can be changed to `// Make the value available in DEBUG builds` (the "fastchecked" is long gone)

Reply:
`instrVal` was moved lower, maybe I can fix it in another PR, because there are many places with this comment. Or we can delete this `forceStack`, I am not sure how useful it is.